### PR TITLE
emacs.pkgs.control-lock: Add package

### PR DESCRIPTION
**Pull request by hadriencaron**
###### Description of changes

Add emacs package that allows to caps-lock like behavior for control key (with cursor change of color)

###### Things done
- Built on platform(s)
  - [X] x86_64-linux

I have just tested it by running 
$ nixos-rebuild switch  -I nixpkgs=/path/to/my/nixpkgs/checkout/nixpkgs

### CODE REVIEW

File: pkgs/applications/editors/emacs/elisp-packages/control-lock/default.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{ lib, stdenv, fetchurl }:
+
+stdenv.mkDerivation {
+  name = "control-lock";
+
+  src = fetchurl {
+    url = "http://emacswiki.org/emacs/download/control-lock.el";
```
Comment (adisbladis) [inline pkgs/applications/editors/emacs/elisp-packages/control-lock/default.nix:7]: This needs to use a version pinned URL.

**Review by adisbladis — CHANGES_REQUESTED** [review]

File: pkgs/applications/editors/emacs/elisp-packages/control-lock/default.nix (hunk 1 above)
Comment (hadriencaron) [reply pkgs/applications/editors/emacs/elisp-packages/control-lock/default.nix:7]: Couldn't find any "versioned" URL.
Seems we do not always require a version (ie: pkgs/applications/editors/emacs/elisp-packages/perl-completion ?)
Or would you be able to suggest a way to create my own ?
Thanks for your help, am new to this !

Comment (adisbladis) [reply pkgs/applications/editors/emacs/elisp-packages/control-lock/default.nix:7]: You can use this mirror https://github.com/emacsmirror/emacswiki.org/tree/master

Comment (adisbladis) [reply pkgs/applications/editors/emacs/elisp-packages/control-lock/default.nix:7]: And BTW, the pearl-completion derivation should also be fixed.
It just didn't happen yet.

Comment (hadriencaron) [reply pkgs/applications/editors/emacs/elisp-packages/control-lock/default.nix]: Understood -- thank you!

**Comment by jian-lin** [comment]
Hi @hadriencaron, would you like to become a maintainer of this `control-lock` package?

We may remove elisp packages from EmacsWiki with an empty `meta.maintainers` or mark them insecure in the future.

**Pull request merged**